auth.go

```diff
@@ -1,36 +1,66 @@
 package main
 
 import (
-	"errors"
-
-	"github.com/gin-gonic/gin"
+	"log/slog"
+	"net/http"
 )
 
-// TODO: Implement proper auth. This function takes a gin context and returns
-// the authenticated username, or error on auth fail.
-func (av *Archiiv) auth(c *gin.Context) (string, error) {
-	return c.MustGet(gin.AuthUserKey).(string), nil
+func getSessionToken(r *http.Request) string {
+	return r.Header.Get("authorization")
 }
 
-func (av *Archiiv) authFile(user string, f *File, p uint8) error {
-	if user == "root" {
-		return nil
-	}
+func getUsername(r *http.Request) string {
+	// TODO
+	return ""
+}
 
-	if user == "marian" {
-		return errors.New("lol web dev")
-	}
+func validateToken(secret, token string) bool {
+	// TODO validate session token
+	return true
+}
+
+func login(name, pwd string, userStore userStorer) (ok bool, token string) {
+	// TODO generate session token
+	token = ""
+	ok = true
+	return
+}
 
-	// NOTE(mrms): the default value if permission is not found in the dict is
-	// 0, which is coincidentaly the same as no permissions.
-	perm, exists := f.Perms[user]
-	if !exists {
-		perm = f.Perms["pub"]
-	}
+func handleLogin(secret string, log *slog.Logger, userStore userStorer) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		name := r.FormValue("name")
+		pwd := r.FormValue("pwd")
 
-	if perm&p == 0 {
-		return errors.New("permission denied")
-	}
+		ok, token := login(name, pwd, userStore)
+
+		if ok {
+			log.Info("New login", "user", name)
+			encodeOK(w, struct {
```

Comment:
Any reason for using a struct instead of just `map[str]any{ "token": token }`?

Reply:
Just that we already have a fancy encodeOK function that takes a struct

Reply:
The function can still take `any`, however I think that in this case using a simple map is easier. But both solutions are OK.